frontend/lib/resolution/scope-queries.cpp

```diff
@@ -2244,6 +2262,306 @@ void resolveUsesAndImportsInScope(Context* context, const Scope* scope) {
   return;
 }
 
+// adds all elements in intersect(a,b) into the set dst
+static void doSetIntersect(const std::set<UniqueString>& a,
```

Comment:
This and `doSetDifference` feel like utility methods that don't belong in this file

Reply:
Hmm... what file do you think we would put them in? They are pretty short & I'm not so sure we will be facing these operations on sets of UniqueString in other situations.

Reply:
I added a TODO comment for now.

Reply:
That's true, they probably won't be used elsewhere. I was worried about adding more to this file that isn't scope-related logic, but since these are so small and are only ever used directly below I think it's fair enough.